frontend/app/services/PayPalService.scala

```diff
@@ -1,36 +1,43 @@
 package services
 
+import actions.AuthRequest
+import com.gu.identity.play.IdMinimalUser
+import com.gu.okhttp.RequestRunners
+import com.gu.paypal.PayPalConfig
+import com.gu.touchpoint.TouchpointBackendConfig.BackendType
 import com.netaporter.uri.Uri.parseQuery
 import com.typesafe.scalalogging.LazyLogging
 import configuration.Config
 import controllers.PayPal.{PayPalBillingDetails, Token}
 import controllers.routes
 import okhttp3.{FormBody, OkHttpClient, Request, Response}
-import play.api.mvc.RequestHeader
+import play.api.mvc.AnyContent
+import utils.TestUsers
 
 object PayPalService extends LazyLogging {
 
   // The parameters sent with every NVP request.
-  private val defaultNVPParams = Map(
-    "USER" -> Config.paypalUser,
-    "PWD" -> Config.paypalPassword,
-    "SIGNATURE" -> Config.paypalSignature,
-    "VERSION" -> Config.paypalNVPVersion)
+  private def defaultNVPParams(requestConfig : PayPalConfig) = Map(
+    "USER" -> requestConfig.user,
+    "PWD" -> requestConfig.password,
+    "SIGNATURE" -> requestConfig.signature,
+    "VERSION" -> requestConfig.NVPVersion)
 
   // Takes a series of parameters, send a request to PayPal, returns response.
-  private def nvpRequest(params: Map[String, String]) = {
+  private def nvpRequest(params: Map[String, String], user : IdMinimalUser) = {
+    logger.info("Is test user = " + TestUsers.isTestUser(user))
+    val requestConfig = TouchpointBackend.forUser(user).payPalConfig
```

Comment:
You won't need to hand down the `user`, or derive the `TouchpointBackend.forUser(user).payPalConfig`, if the paypal config is already passed to this class in it's constructor!